integration/lerna-publish-canary.test.js

```diff
@@ -40,3 +41,42 @@ Successfully published:
  - package-3@1.0.1-alpha.0+SHA
 `);
 });
+
+test("lerna publish --canary --no-git-reset leaves the working tree dirty", async () => {
+  const { cwd } = await cloneFixture("normal");
+  const args = ["publish", "--canary", "--yes", "--no-git-reset"];
+
+  await gitTag(cwd, "v1.0.0");
+  await commitChangeToPackage(cwd, "package-1", "change", { change: true });
+
+  const { stdout } = await cliRunner(cwd, env)(...args);
+  expect(stdout).toMatchInlineSnapshot(`
+
+Found 3 packages to publish:
+ - package-1 => 1.0.1-alpha.0+SHA
+ - package-2 => 1.0.1-alpha.0+SHA
+ - package-3 => 1.0.1-alpha.0+SHA
+
+Successfully published:
+ - package-1@1.0.1-alpha.0+SHA
+ - package-2@1.0.1-alpha.0+SHA
+ - package-3@1.0.1-alpha.0+SHA
+`);
+
+  const { stdout: statusStdOut } = await gitStatus();
```

Comment:
I'm not sure we need an entire helper just to run `git status`, and in any case, a better snapshot would be tightly focused on _just_ the modified files, which is already used [elsewhere](https://github.com/lerna/lerna/blob/b69a7282b6841f2b40f94e6a7df5ecf565be97a1/commands/publish/__tests__/git-checkout.test.js#L15):
```suggestion
  const modified = await execa.stdout("git", ["ls-files", "--modified"], { cwd });
```

`modified` in this case is just a newline-separated list of relative file paths, easily snapshotted inline.